Datadog Incident Management guide

bernardjkim opened the pull request:
This PR adds a documentation guide for the Datadog Incident Management plugin that will be added with https://github.com/gravitational/teleport/pull/46271.

bernardjkim committed:
Datadog Incident Management docs

bernardjkim committed:
Backtick

bernardjkim committed:
Merge branch 'master' into bernard/datadog-plugin-doc

bernardjkim committed:
Update datadog config example

bernardjkim committed:
Update docs

bernardjkim committed:
Omit auto approvals

bernardjkim committed:
Omit helm reference

bernardjkim committed:
Fix lint

bernardjkim committed:
Merge branch 'master' into bernard/datadog-plugin-doc

r0mant reviewed (commented):
@bernardjkim Couple of comments:

- Can the guide be geared towards hosted version of the plugin rather than self-hosted? Similar to [Opsgenie](https://goteleport.com/docs/admin-guides/access-controls/access-request-plugins/opsgenie/) or [ServiceNow](https://goteleport.com/docs/admin-guides/access-controls/access-request-plugins/servicenow/). That's one less configuration to support.
- You need to update [config.js](https://github.com/gravitational/teleport/blob/master/docs/config.json) to add the new entry to table of contents on the left nav bar.

bernardjkim committed:
Focus on hosted version

bernardjkim committed:
Update config.json

hugoShaka commented on docs/pages/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog.mdx:
```diff
@@ -0,0 +1,160 @@
+Role Access Requests as Datadog incidents, notify the appropriate on-call team,
+and approve or deny the requests via Teleport.
+
+<Details title="This integration is hosted on Teleport Enterprise (Cloud)" open={false}>
```
It's not clear if this is a cloud-only feature, enteprise-only, or can still be done by running the self-hosted plugin (we do have [the main function here](https://github.com/gravitational/teleport/blob/master/integrations/access/datadog/cmd/teleport-datadog/main.go) for the standalone executable).

hugoShaka commented on docs/pages/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog.mdx:
```diff
@@ -0,0 +1,160 @@
+
+![Datadog enrollment](../../../../img/enterprise/plugins/datadog/datadog-enrollment.png)
+
+If the recipient is a Datadog team, the team name will be added to the Datadog incident
```
> If the recipient is a Datadog team

Can the recipient be something else than a team? If so it would be useful to define supported recipient types.

hugoShaka commented on docs/pages/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog.mdx:
```diff
@@ -0,0 +1,160 @@
+![Datadog enrollment](../../../../img/enterprise/plugins/datadog/datadog-enrollment.png)
+
+If the recipient is a Datadog team, the team name will be added to the Datadog incident
+teams attribute. The plugin will not manually send notifications to members of the
+team. Instead you will want to navigate to **Incidents -> Settings -> Notifications -> Rules**
+and add a notification rule for when an incident with the matching teams attribute
+is created or updated.
```
```suggestion
teams attribute.

The plugin creates incidents and labels them with the recipient team. By default, the team members won't get notified, you must configure Datadog notification rules to page the team members. To do so, navigate to **Incidents -> Settings -> Notifications -> Rules** and create a new rule matching alerts labeled with the recipient team.
```

hugoShaka commented on examples/resources/plugins/teleport-datadog.toml:
If we only document the hosted version of the plugin, is this file used?

hugoShaka commented on docs/pages/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog.mdx:
````diff
@@ -0,0 +1,160 @@
+$ tsh request create --roles=dbadmin --reviewers=alice@example.com,ivan@example.com
+```
+
+If an Access Request includes suggested reviewers, the Datadog Incident Management
+plugin will add these to the list of users to notify.
````
What's the format of sugested reviewrs? emails? do we also support other team names? Can datadog users have usernames that are not emails?

bernardjkim committed:
Address feedback

- Separate guide for hosted vs self-hosted
- Include jq prerequisite
- Smaller screenshots
- Clarify acceptable recipient types
- Clarify acceptable reviewer input

bernardjkim commented:
Roman suggested we document both the hosted and self-hosted guide. Let me know if y'all think having two separate guides is just more confusing, or if we should try to merge them into one guide.

r0mant commented on docs/config.json:
```diff
@@ -1710,6 +1710,16 @@
+      "permanent": true
+    },
+    {
+      "source": "/access-controls/access-request-plugins/ssh-approval-datadog/",
+      "destination": "/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog/",
```
Not sure why these have "ssh" in the page names, I'm guessing it's an artifact of old guides or something, I'd remove it as it doesn't make sense.
```suggestion
      "source": "/access-controls/access-request-plugins/datadog-self-hosted/",
      "destination": "/admin-guides/access-controls/access-request-plugins/datadog-self-hosted/",
```

r0mant commented on docs/pages/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog.mdx:
```diff
@@ -0,0 +1,381 @@
+
+## Step 1/9. Create Datadog team
+
+To demonstrate the Datadog Incident Management plugin, create a team in Datadog.
+In the Datadog web UI, navigate to the **Teams** tab and click on **New Team** to
+create a new Datadog team. Name it "Teleport Access".
+
+We will configure the Datadog Incident Management plugin to route certain access
+requests to the "Teleport Access" team.
```
Some of these steps are exactly the same between hosted and self-hosted guides, can we move them out into includes?

r0mant reviewed (approved):
lgtm with a couple of comments

Do not forget to update these guides once you've added auto-approval.

r0mant commented on docs/pages/admin-guides/access-controls/access-request-plugins/ssh-approval-datadog.mdx:
````diff
@@ -0,0 +1,381 @@
+Here is an example of a `role_to_recipients` map:
+
+```toml
+[role_to_recipients]
+"*" = ["admin-team"]
+"dev" = ["dev-team", "admin-team"]
+"dba" = ["user@example.com"]
````
Role to recipients is deprecated in favor of notification routing rules. Let's not use this anymore.

r0mant commented on examples/resources/plugins/teleport-datadog.toml:
```diff
@@ -0,0 +1,48 @@
+# Datadog Severity specifies the severity level of incidents.
+severity = "SEV-3"
+
+[role_to_recipients]
+# Map roles to recipients.
+#
+# Provide datadog user_email/team recipients for access requests for specific roles.
+# role.suggested_reviewers will automatically be treated as additional email recipients.
+# "*" must be provided to match non-specified roles.
+#
+# "dev" = ["dev-team"]
+# "*" = ["cloud@email.com", "cloud-team"]
```
As mentioned above, let's stick to notification routing rules instead of this, this is legacy configuration.

hugoShaka replied:
You still need to put the section with the catchall recipient for the plugin to start. We can edit the comment to say that AMRs are a better way of setting additional recipients.

mmcallister commented on docs/img/enterprise/plugins/datadog/diagram.png:
Looks like this is using the old Teleport logo

bernardjkim replied:
Not sure how to update the Teleport logo on these diagrams. I'll just omit the diagram for now and defer this change to a later PR. We'll probably want to update all the outdated diagrams in a batch.

mmcallister replied:
Thanks Bernard! I have a PR open to work on updating all of the images using the old logo. Just didn't want to throw more in there 😁

hugoShaka commented:
> Roman suggested we document both the hosted and self-hosted guide. Let me know if y'all think having two separate guides might be more confusing, or if we should try to merge them into one guide.

I think we'll need to merge guides, especially if we do more dual "self-hosted/teleport-hosted" guides like this. The navifation is already a mess and this won't scale very well if we do this for every plugin:
<img width="253" alt="image" src="https://github.com/user-attachments/assets/1ff56910-d881-44bc-b674-ed0b516ea557">

Sorry for giving you extra work :/ (the PR content is very good though)

hugoShaka commented on docs/pages/admin-guides/access-controls/access-request-plugins/datadog-hosted.mdx:
```diff
@@ -0,0 +1,204 @@
+---
+title: Access Requests with Datadog Incident Management (Hosted)
```
I think the "Hosted" vs "Self-hosted" working can be improved. It's not clear for users what a hosted plugin is. Also self hosted is technically hosted by yourself, so they are not opposed concepts.

We would benefit for picking clear and well-defined terms, especially as one of our proposed Q4 goals is to align plugins in terms of feature (all access plugins should support both hosted and self-hosted modes).

WDYT of:
- Standalone vs embedded
- Standalone vs integration
- Self-hosted vs Teleport-Hosted

cc @r0mant

hugoShaka commented on docs/pages/admin-guides/access-controls/access-request-plugins/datadog-hosted.mdx:
```diff
@@ -0,0 +1,204 @@
+- Either a Linux host or Kubernetes cluster where you will run the Datadog plugin.
+
+- (!docs/pages/includes/tctl.mdx!)
+
```
Editor-like rights in Teleport are also required (ability to create roles and grant them to users).

hugoShaka commented on docs/pages/admin-guides/access-controls/access-request-plugins/datadog-hosted.mdx:
```diff
@@ -0,0 +1,204 @@
+
+The fallback recipient will be the default recipient for notifications. The recipient
+can be a Datadog user email, or a Datadog team handle. You can configure more custom
+notification routing rules afterwards using Access Monitoring Rules.
```
A user not familiar with Teleport will ask themselves "what is an AMR?", so this would benefit from a link. We don't have a nice "What is an AMR" page yet, so the most apropriate would be the AMR guide you already linked in the "Next steps".

bernardjkim commented:
Okay, to limit the scope of this PR and get it merged by next week, I'll only include the "teleport-hosted" guide for this plugin. I agree that we should also include a "self-hosted" guide, but I think we want to prioritize "teleport-hosted".

Some things we might want to address before including a second "self-hosted" guide:
- The parent [Just-in-Time Access Request Plugins](https://goteleport.com/docs/admin-guides/access-controls/access-request-plugins/) page documents the plugins as "Teleport Cloud hosted" and "Self-hosted". We should decide on wording and make it consistent everywhere in the docs.
- In the parent page, these docs were originally intended only for "self-hosted" plugin guides. But it seems like it has now become a mix of "self-hosted" and "teleport-hosted". Suggestions to address this inconsistency:
  - We should organize the guides to have separate sections for "self-hosted" vs "teleport-hosted".
  - Alternatively, we remove all "teleport-hosted" guides from the documentation. Instead we improve the WebUI to include all the needed documentation and guide the user to a successful setup.
<img width="1192" alt="Screenshot 2024-10-11 at 10 02 16 AM" src="https://github.com/user-attachments/assets/7f9d82cd-bad7-4cac-9682-daa1382b0c61">

bernardjkim committed:
Update docs

- Remove self-hosted guide
- Remove outdated diagram
- Include editor role prerequisite
- Include AMR link
- Include auto approvals setup

bernardjkim committed:
Remove self-hosted examples

bernardjkim committed:
Fix lint

bernardjkim committed:
Change datadog_auto_approvals -> teleport.dev/schedules

bernardjkim committed:
Use relative link

bernardjkim committed:
Merge branch 'master' into bernard/datadog-plugin-doc
